src/fileprovider.cs

```diff
@@ -910,5 +984,290 @@ interface NSFileProviderReplicatedExtension : NSFileProviderEnumerating {
 
 		[Export ("materializedItemsDidChangeWithCompletionHandler:")]
 		void MaterializedItemsDidChange (Action completionHandler);
+
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Unavailable (PlatformName.MacCatalyst)]
+		[NoWatch, NoTV, NoiOS, Mac (11,3)]
+		[Export ("pendingItemsDidChangeWithCompletionHandler:")]
+		void PendingItemsDidChange (Action completionHandler);
+	}
+
+	interface INSFileProviderDomainState { }
+
+	[Advice ("This API is not available when using UIKit on macOS.")]
+	[Unavailable (PlatformName.MacCatalyst)]
+	[NoWatch, NoTV, NoiOS, Mac (11,3)]
+	[Protocol]
+	interface NSFileProviderDomainState {
+
+		[Abstract]
+		[Export ("domainVersion")]
+		NSFileProviderDomainVersion DomainVersion { get; }
+
+		[Abstract]
+		[Export ("userInfo", ArgumentSemantic.Strong)]
+		NSDictionary UserInfo { get; }
+	}
+
+	[Advice ("This API is not available when using UIKit on macOS.")]
+	[Unavailable (PlatformName.MacCatalyst)]
+	[NoWatch, NoTV, NoiOS, Mac (11,3)]
+	[Flags]
+	[Native]
+	public enum NSFileProviderDomainTestingModes : ulong {
+		AlwaysEnabled = 1uL << 0,
+		Interactive = 1uL << 1,
+	}
+
+	[Advice ("This API is not available when using UIKit on macOS.")]
+	[Unavailable (PlatformName.MacCatalyst)]
+	[NoWatch, NoTV, NoiOS, Mac (11,3)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSFileProviderDomainVersion : NSSecureCoding {
+
+		[Export ("next")]
+		NSFileProviderDomainVersion Next { get; }
+
+		[Export ("compare:")]
+		NSComparisonResult Compare (NSFileProviderDomainVersion otherVersion);
+	}
+
+	[Unavailable (PlatformName.MacCatalyst)]
+	[NoWatch, NoTV, NoiOS, Mac (11,3)]
+	[Native, Advice ("This API is not available when using UIKit on macOS.")]
+	public enum NSFileProviderTestingOperationType : long {
+		Ingestion = 0,
+		Lookup = 1,
+		Creation = 2,
+		Modification = 3,
+		Deletion = 4,
+		ContentFetch = 5,
+		ChildrenEnumeration = 6,
+		CollisionResolution = 7,
+	}
+
+	interface INSFileProviderTestingOperation : global::ObjCRuntime.INativeObject { }
```

Comment:
why was `: global::ObjCRuntime.INativeObject { }` needed ?

Reply:
Because it is used inside `NSDictionary<INSFileProviderTestingOperation, NSError>` and the info available just yet.